corehq/apps/export/system_properties.py

```diff
@@ -55,140 +55,147 @@
         label='app_id',
-        item=ExportItem(path=[PathNode(name='app_id')]),
+        item=ExportItem(path=[PathNode(name='app_id')], datatype='string'),
         is_advanced=True,
         help_text=_("The ID of the app that this form is part of")
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_APP],
         label='build_id',
-        item=ExportItem(path=[PathNode(name='build_id')]),
+        item=ExportItem(path=[PathNode(name='build_id')], datatype='string'),
         is_advanced=True,
         help_text=_("The ID of the published app that this form is part of")
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_APP],
         label='@version',
-        item=ExportItem(path=[
-            PathNode(name='form'), PathNode(name='@version')
-        ]),
+        item=ExportItem(
+            path=[PathNode(name='form'), PathNode(name='@version')],
+            datatype='string',
+        ),
         is_advanced=True,
         help_text=_("The version of the app in which this form was last updated prior to being published")
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_SERVER],
         label="state",
-        item=ExportItem(path=[PathNode(name="doc_type")]),
+        item=ExportItem(path=[PathNode(name="doc_type")], datatype='string'),
         is_advanced=True
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_SERVER],
         label="last_sync_token",
-        item=ExportItem(path=[PathNode(name="last_sync_token")]),
+        item=ExportItem(path=[PathNode(name="last_sync_token")], datatype='string'),
         is_advanced=True,
         help_text=_("The ID of the last sync on the phone that occurred prior to submitting this form.")
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_SERVER],
         label="partial_submission",
-        item=ExportItem(path=[PathNode(name="partial_submission")]),
+        item=ExportItem(path=[PathNode(name="partial_submission")], datatype='boolean'),
         is_advanced=True,
         help_text=_("True if the form was a partial submission, False otherwise.")
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_SERVER],
         label="received_on",
-        item=ExportItem(path=[PathNode(name="received_on")]),
+        item=ExportItem(path=[PathNode(name="received_on")], datatype='datetime'),
         selected=True,
         help_text=_("The time at which the server receive this form submission"),
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_SERVER],
         label="edited_on",
-        item=ExportItem(path=[PathNode(name="edited_on")]),
+        item=ExportItem(path=[PathNode(name="edited_on")], datatype='datetime'),
         is_advanced=True,
         help_text=_("The time at which this form was last edited."),
     ),
     ExportColumn(
         tags=[PROPERTY_TAG_SERVER],
         label="submit_ip",
-        item=ExportItem(path=[PathNode(name="submit_ip")]),
+        item=ExportItem(path=[PathNode(name="submit_ip")], datatype='string'),
```

Comment:
nit: could these datatypes be constants?

Reply:
good idea. https://github.com/dimagi/commcare-hq/pull/19987/commits/9c7d9cbd58a5924e45c0287381a426dc4bea79dd. i wasn't totally sure where they should live but for now am keeping them in UCR land. also didn't bother fixing all references to them because... there are a lot.